### PR TITLE
BFD-2737: Adjust github hook installation and move/rename githook secrets file

### PR DESCRIPTION
**Pull request by lsmitchell**
**JIRA Ticket:**
[BFD-2737](https://jira.cms.gov/browse/BFD-2737)

**User Story or Bug Summary:**

As a BFD contributor, I want the pre-commit and pre-push hooks to be sane so that I can use them with sanity.

---

### What Does This PR Do?

- Moves the git hooks into their own files instead of shoved into the parent pom
  - These files will be located at .github/scripts
- Replaces the plugin for installing git hooks we had with a new one that supports reading from files
  - The new plugin also works on build, instead of needing to initialize, so the first build from the root directory will install the git hooks, simplifying the onboarding slightly
- Moves the .secrets file into .github/resources and renamed it to .sensitive-files to avoid naming conflicts with things that need a .secrets file (and better explain what it is for)
  - Adjusted the commit hook script to work with the new location/name

### What Should Reviewers Watch For?

If you're reviewing this PR, please check for these things in particular:

* Verify all PR security questions and checklists have been completed and addressed.

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then **you must** supply a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence here: **N/A**

    * Does this PR add any new software dependencies? 
      * [x] Yes
      * [ ] No
    * Does this PR modify or invalidate any of our security controls?
      * [ ] Yes
      * [x] No
    * Does this PR store or transmit data that was not stored or transmitted before?
      * [ ] Yes
      * [x] No

* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons?
      * [x] Yes
      * [ ] No

### What Needs to Be Merged and Deployed Before this PR?

This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

* N/A

### Submitter Checklist

I have gone through and verified that...:

* [x] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
* [x] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [x] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [x] The data dictionary has been updated with any field mapping changes, if any were made.
* [x] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [x] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [x] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
* [x] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

**Comment by lsmitchell** [comment]
@StewGoin new plugin is:

https://github.com/rudikershaw/git-build-hook
https://mvnrepository.com/artifact/com.rudikershaw.gitbuildhook/git-build-hook-maven-plugin

Replacing:

https://github.com/phillipuniverse/githook-maven-plugin
https://mvnrepository.com/artifact/io.github.phillipuniverse/githook-maven-plugin

Didn't see any damning dependencies or anything in the new plugin. (Lots of its dependencies including slf4j are test scoped)
https://github.com/rudikershaw/git-build-hook/blob/develop/pom.xml

**Comment by lsmitchell** [comment]
If accepted as-is, I'll update the wiki to note that to install the git hooks you just need to do a build from the apps directory instead of mvn initialize. With a fresh repo I pulled down, the .git folder is already added; there should be no reason to need initialize anymore with this change.

File: .github/scripts/pre-commit.sh [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  # make list of files to be committed by printing out a newline
+  # separated list of files staged for commit to a temporary file
+  # and reading it as an array
+  tmpfile=$(mktemp)
+  git diff --cached --name-only --diff-filter=ACM > $tmpfile
+  IFS=$'\n' read -d '' -r -a commits < $tmpfile
+
+  # for all files to be committed which are a secret file,
+  # grep for the header "$ANSIBLE_VAULT;1.1;AES256"
+  # if it's not there, abort.
```
Comment (JeremyIR) [inline .github/scripts/pre-commit.sh:19]: This is really nitpicky, but the commend on lines 10-12 doesnt have a period at the end.

Comment (JeremyIR) [reply .github/scripts/pre-commit.sh:19]: comment*

**Comment by JeremyIR** [comment]
Looks good to me aside from consistency with comments ending in a period or not, if it matters that much

**Review by JeremyIR — DISMISSED** [review]

File: .github/scripts/pre-commit.sh (hunk 1 above)
Comment (lsmitchell) [reply .github/scripts/pre-commit.sh]: True, just copied it from the pom, but I'll fix this.

**Review by brianburton — APPROVED** [review]

**Comment by lsmitchell** [comment]
Just noticed the renamed .secrets file didn't make it into the commit, since it was being .gitignored; fixed the gitignore file and added the newly rehomed/rename secrets file.

**Comment by JeremyIR** [comment]
@lsmitchell any advice on how to best test this? I know I have to clone your branch and install the plugin but any further guidance would be appreciated

**Comment by lsmitchell** [comment]
If you grab this branch you can test it by doing the following:
1. Switch to the branch
2. Go to <bfd-repo>/.git/hooks and delete the pre-commit and pre-push hooks that are there
3. Maven build from the top level in apps (can skip tests)
4. Check that the hooks were installed into <bfd-repo>/.git/hooks

I did the following to test the pre-push hook is working:
once the hooks are installed go and change of the files listed in the sensitive-files (such as ops/terraform/services/base/values/test.eyaml) and remove the first line or replace everything with some plaintext, then try to commit. You should see it fail to commit due to the file being in plaintext. You can also see the pre-commit hook messages will work fine when committing the change.

**Comment by JeremyIR** [comment]
Verifying on my end that the functionality works as properly. LGTM

**Review by JeremyIR — APPROVED** [review]

**Comment by StewGoin** [comment]
No concerns from me, proceed as planned.

**Review by brianburton — APPROVED** [review]

**Pull request merged**